### ISSUE TITLE
Installtool broken after update to 4.6.11

### ISSUE THREAD

**Issue by e9li** [contao/contao#238]
**Affected version(s)**
4.6.11

**Description**  
After updating to 4.6.11 the install tool throws "There is no column with name 'path(768)' on table 'tl_files'." The page itself works, just the install tool is broken.

**How to reproduce** 
1. Update to Contao 4.6.11
1. Open the install tool

**Comment by netzarbeiter** [comment]
Same issue with Contao 4.4.31.

The problem is the update from doctrine/dbal 2.9.0 to 2.9.1.

Adding the following code to the composer.json and do a `composer.phar update` did the trick.

```
"conflict": {
        "doctrine/dbal": "2.9.1"
 }
```

**Comment by contaoacademy** [comment]
Same error here on 4.4.31. Thx @netzarbeiter for the fix.

**Comment by Shoekrates** [comment]
Same error over here while updating from 4.4.30 to 4.4.31 with Contao Manager 1.1.2 on PHP 7.2.11 (Hoster all-inkl.). Adding @netzarbeiter conflict information to composer.json did the trick. 
A general question: Is this conflict information something temporary that gets deleted automatically with the next update, do we have to change it manually or does it have to stay there „forever“?

**Comment by e9li** [comment]
Thx @netzarbeiter for the fix.

**Comment by fritzmg** [comment]
> A general question: Is this conflict information something temporary that gets deleted automatically with the next update, do we have to change it manually or does it have to stay there „forever“?

It stays forever, if you do not delete it yourself. However, since you have only conflicted `2.9.1` of the package, any version above (or below) that can still be installed.

**Comment by aschempp** [comment]
Can someone post the changes the install tool wants to apply?

**Comment by e9li** [comment]
In my cases there where not changes. After the update I just wanted to check if some updates where needed. After the fix of @netzarbeiter and clearing the cache the install tool worked as expected.

**Comment by leofeyer** [comment]
https://github.com/doctrine/dbal/pull/3392#issuecomment-447280144

**Comment by qzminski** [comment]
For me it failed with exception:

```
There is no column with name 'path(333)' on table 'tl_files'.
```

… but I am updating Contao 3.5 database to 4.4. @netzarbeiter fix worked for me though :+1:

**Comment by stefanschleich** [comment]
Failed too after updating from 4.6.8 to 4.6.11:

    [2018-12-14 12:06:00] request.INFO: Matched route "contao_install". {"route":"contao_install","route_parameters":{"_route":"contao_install","_scope":"backend","_token_check":true,"_controller":"Contao\\InstallationBundle\\Controller\\InstallationController::installAction"},"request_uri":"***/contao/install","method":"HEAD"} []
    [2018-12-14 12:06:00] request.INFO: Matched route "contao_install". {"route":"contao_install","route_parameters":{"_route":"contao_install","_scope":"backend","_token_check":true,"_controller":"Contao\\InstallationBundle\\Controller\\InstallationController::installAction"},"request_uri":"***/contao/install","method":"GET"} []
    [2018-12-14 12:06:00] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Schema\SchemaException: "There is no column with name 'path(768)' on table 'tl_files'." at /***/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/SchemaException.php line 82 {"exception":"[object] (Doctrine\\DBAL\\Schema\\SchemaException(code: 30): There is no column with name 'path(768)' on table 'tl_files'. at /***/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/SchemaException.php:82)"} []

**Comment by pandroid** [comment]
Same here after update from 4.6.8 to 4.6.10

`[2018-12-14 13:03:55] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Schema\SchemaException: "There is no column with name 'path(768)' on table 'tl_files'." at /***/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/SchemaException.php line 82 {"exception":"[object] (Doctrine\\DBAL\\Schema\\SchemaException(code: 30): There is no column with name 'path(768)' on table 'tl_files'. at /***/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/SchemaException.php:82)"} []`

**Comment by b-fender** [comment]
same here, Contao 4.4.31, PHP 7.2, MySQL 5.5.5-10.0.36-MariaDB:
app.CRITICAL: An exception occurred. {"exception":"[object] (Doctrine\\DBAL\\Schema\\SchemaException(code: 30): There is no column with name 'path(333)' on table 'tl_files'. at ........vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/SchemaException.php:82)

**Comment by leofeyer** [comment]
Please stop spamming the people watching this issue with useless "same here" comments that provide no further information. We have already acknowledged the issue and are now waiting for a statement by the Doctrine maintainers (see https://github.com/doctrine/dbal/issues/3409).

**Comment by ghost** [comment]
Hi 

How can I launch a composer update without access to ssh console? I'm using contao manager only.

kind regards

**Comment by aschempp** [comment]
Using the Contao Manager, the maintenance screen has a composer install/update option.

**Comment by ghost** [comment]
Thanks

**Comment by leofeyer** [comment]
@contao/developers I would solve this by only allowing `doctrine/dbal` < 2.9 in Contao 4.4 and by adapting to the new syntax and only allowing `doctrine/dbal` >= 2.9 in Contao 4.6. Any objections?

**Comment by Toflar** [comment]
Any reason why not adapting the new syntax and requiring `^2.9.1` in both versions? I don't think 2.8 is maintained anymore and we need to have maintained versions in Contao 4.4 until December 2021.

**Comment by leofeyer** [comment]
We cannot, because `doctrine/dbal` requires PHP 7.1 as of version 2.6.

**Comment by Toflar** [comment]
I see. Well, at some point I guess we have to increase the minimum required PHP version for Contao 4.4 as well because the underlying components require to do so. Let's say Doctrine contains a security vulnerability which also affects us, we must be able to update in that case so we can provide a stable and secure Contao 4.4 until the end of 2021. I'm pretty sure this won't be possible without raising the minimum required PHP version at some point. Maybe not today but it's still a long way to go until 2021.

So if you want to go that way, I'm okay with it. But at the same time I just want to make it clear that I think the point in time where we have no other option than raising the minimum required PHP version will come and we might as well do it now. I think we should at least consider it. To be fair, I think the majority of people run Contao 4.4 on PHP 7.x anyway.

**Comment by leofeyer** [comment]
Although I generally agree, I think you are wrong about "we need to have maintained versions in Contao 4.4 until December 2021".

Maintaining an LTS version means "providing a certain software state and promising to fix bugs for a certain period of time". However, it does not mean that we need to have only maintained versions of the third-party packages – just like not every Linux distribution only ships maintained versions of the bundled software.

If a package has a bug or a security issue, having a maintained version surely helps, because all you have to do is to update. But you might as well fork the package and provide a patched version (which is what most Linux distributions do).

Therefore I am strongly against raising the minimum PHP version in a bugfix release unless absolutely necessary!

**Comment by fritzmg** [comment]
> To be fair, I think the majority of people run Contao 4.4 on PHP 7.x anyway.

Especially since major shared hosters now have removed PHP 5.6.

**Comment by leofeyer** [comment]
> Especially since major shared hosters now have removed PHP 5.6.

First of all, only a few really have. And for every company that has, there are 10 companies that are now advertising "guaranteed PHP 5.6 support at least until 2021" and similar things. Just google for "php 5.6 hosting" and see for yourself.

**Comment by leofeyer** [comment]
Fixed in Contao 4.6 in 116e66dc4bc7acb2e7c28bf6a3047cbf240c2eb4. Note that Doctrine still needs to fix https://github.com/doctrine/dbal/issues/3414 so index length changes are detected correctly.

**Comment by leofeyer** [comment]
Fixed in Contao 4.4 in 9cbadc0b9cfdb608cdfd883fe6944ab653e1ab94.
